src/dali/plugin/pair_converter/csv_reader/kraken_csv_pricing.py

```diff
@@ -0,0 +1,205 @@
+# Copyright 2022 macanudo527
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Kraken CSV format:
+
+import logging
+from csv import reader
+from datetime import datetime, timedelta, timezone
+from io import BytesIO
+from itertools import repeat
+from json import JSONDecodeError
+from multiprocessing.pool import ThreadPool
+from typing import Any, Dict, Iterable, List
+from zipfile import ZipFile
+
+import requests
+from requests.models import Response
+from requests.sessions import Session
+from rp2.logger import create_logger
+from rp2.rp2_decimal import RP2Decimal
+
+from dali.historical_bar import HistoricalBar
+
+# Google Drive parameters
+_ACCESS_NOT_CONFIGURED: str = "accessNotConfigured"
+_ERROR: str = "error"
+_ERRORS: str = "errors"
+_FILES: str = "files"
+_GOOGLE_API_KEY: str = "AIzaSyBPZbQdzwVAYQox79GJ8yBkKQQD9ligOf8"
+_GOOGLE_APIS_URL: str = "https://www.googleapis.com/drive/v3/files"
+_ID: str = "id"
+_KRAKEN_FOLDER_ID: str = "1aoA6SKgPbS_p3pYStXUXFvmjqShJ2jv9"
+_MESSAGE: str = "message"
+_REASON: str = "reason"
+
+# Google Drive URL Params
+_ALT: str = "alt"
+_API_KEY: str = "key"
+_CONFIRM: str = "confirm"
+_MEDIA: str = "media"
+_QUERY: str = "q"
+
+_MS_IN_SECOND: int = 1000
+
+
+class KrakenCsvPricing:
+
+    __KRAKEN_OHLCVT: str = "Kraken.com_CSVOHLCVT"
+
+    __TIMEOUT: int = 30
+    __THREAD_COUNT: int = 6
+
+    __TIMESTAMP_INDEX: int = 0
+    __OPEN: int = 1
+    __HIGH: int = 2
+    __LOW: int = 3
+    __CLOSE: int = 4
+    __VOLUME: int = 5
+    __TRADES: int = 6
+
+    __DELIMITER: str = ","
+
+    def __init__(
+        self,
+        google_api_key: str,
```

Comment:
Does the google_api_key need to be passed in as a parameter? This class already knows what it is (it's `_GOOGLE_API_KEY`). Removing the parameter from the costructor would also allow you to simplify the reflective logic at line 360 in ccxt.py.

Reply:
It's required, unfortunately. This is the user's private key. The one that was embedded was for early testing purposes. We can't keep using the private key since it has d/l limitations. Users will need to get their own and enable Google Drive on it, unfortunately.